heat/core/manipulations.py

```diff
@@ -316,6 +329,169 @@ def concatenate(arrays, axis=0):
             return out
 
 
+def diag(a, offset=0):
+    """
+    Extract a diagonal or construct a diagonal array.
+    See the documentation for `heat.diagonal` for more information about extracting the diagonal.
+
+    Parameters
+    ----------
+    a: ht.DNDarray
+        The array holding data for creating a diagonal array or extracting a diagonal.
+        If a is a 1-dimensional array a diagonal 2d-array will be returned.
+        If a is a n-dimensional array with n > 1 the diagonal entries will be returned in an n-1 dimensional array.
+    offset: int, optional
+        The offset from the main diagonal.
+        Offset greater zero means above the main diagonal, below zero is below the main diagonal.
+
+    Returns
+    -------
+    res: ht.DNDarray
+        The extracted diagonal or the constructed diagonal array
+
+    Examples
+    --------
+    >>> import heat as ht
+    >>> a = ht.array([1, 2])
+    >>> ht.diag(a)
+    array([[1, 0],
```

Comment:
why does these say array instead of tensor?

Reply:
I just realized that we are a little inconsistent in the documentation. Some examples use `array`, some use `tensor`. I will fix mine now but we probably should also make a PR that fixes all of the other functions to be in line with what actually will be displayed by the functions.